pkg/v1/tarball/layer.go

```diff
@@ -68,16 +71,30 @@ func (l *layer) MediaType() (types.MediaType, error) {
 	return types.DockerLayer, nil
 }
 
+// LayerOption applies options to layer
+type LayerOption func(*layer) error
+
+// WithCompressionLevel sets the gzip compression. See `gzip.NewWriterLevel` for possible possible values.
+func WithCompressionLevel(level int) LayerOption {
+	return func(l *layer) error {
+		if level < gzip.HuffmanOnly || level > gzip.BestCompression {
+			return errors.Errorf("invalid compression level: %d", level)
```

Comment:
I've been meaning to drop this package as a dependency (it's only used in a couple places) -- can this just be `fmt.Errorf`?